keps/sig-network/784-kube-proxy-component-config/README.md

```diff
@@ -155,7 +218,19 @@ The config should be considered graduated to beta if it:
 - is well documented. Especially in regard to migrating to it from older versions.
 
 ### Upgrade / Downgrade Strategy
-
+#### kubeadm
+The `init` command will create v1alpha2 kube-proxy configuration for bootstrapping new clusters. 
+The `upgrade` command will mutate the existing v1alpha1 to v1alpha2 kube-proxy configuration. 
+```bash
+kubeadm upgrade
+```
+
+#### kube-proxy
+The existing flags `--config` and `--write-config-to` can be used to convert any existing v1alpha1 to v1alpha2 kube-proxy configuration.
+`--config` can consume and decode any supported version, `--write-config-to` will always write using latest version.
+```bash
+/usr/local/bin/kube-proxy --config old-v1alpha1.yaml --write-config-to new-v1alpha2.yaml
```

Comment:
Will v1alpha1 keep working?
there are millions of kube-proxy out there working for more almost a decade, if we don't make a change that is transparent for the users we are going to break a lot of people for a very small win.

Reply:
If we can cheat and say that in the future new features will be added to v1alpha2/beta/ga but not v1alpha1, then maintaining v1alpha1 support forever-ish is very little work.

(If we feel the need to obey API compat rules more rigorously, or if the tooling effectively forces us to, then we could just add a `V1Alpha2Config *v1alpha2.KubeProxyConfiguration` field to the end of `v1alpha1.KubeProxyConfiguration`, and then v1alpha1 automatically gets support for anything we add to v1alpha2, without us needing to actually do any further maintenance on v1alpha1.)

Reply:
I think the key is that despite the config says it is alpha is not really alpha, so I would prefer we don't break people, at the end of the day I didn't see any issues complaining about the config, most of the complains are for the config vs flag thing. It is mostly us that we recognize the mess we have as you perfectly explain in https://github.com/kubernetes/kubernetes/issues/117909

Is it possible to implement the roundtrip between versions without having to boil the ocean?

Reply:
> most of the complains are for the config vs flag thing

(which can't be fixed without breaking compat, which the new config version gives us the ability to do)

> Is it possible to implement the roundtrip between versions without having to boil the ocean?

The roundtrip between the current version of v1alpha1 and the initial version of v1alpha2 is trivial. I was just worrying about what happens in the future; if we don't want to ever remove v1alpha1, then we should figure out some way to make it not become a maintenance burden.

(Is roundtripping even relevant for config files? For API objects, it needs to be possible so an old client can Get()+modify+Update() without losing information. But people don't do that with config files...)

Reply:
yeah, I don't think we need to overcomplicate it, I think that we should keep  alphav1 to keep working as always, new options will go only to alphav2,  and if someone wants something new features,  he must switch to alphav2